### ISSUE TITLE
compiler: Maximum recursion depth & maximum regular expression pattern length 

### ISSUE THREAD

**Issue by danielle-mustillo** [google/mtail#471]
Hi mTail team,

We are seeing the following errors upgrading from rc34 to anything above rc34:

`Expression exceeded maximum recursion depth.`

and 

`sipp.mtail:90:1-52: Exceeded maximum regular expression pattern length of 1024 bytes with 2820.`

(the later shows up when we resolve the former issue with a patch I'll describe later)

We have a fairly long mtail expression (couple hundred lines) to parse a fairly large CSV. We don't have control over the source program output so we can't really change the mtail expression. We also want to collect every single value so reducing what is captured by our regexs is not a solution for us. 

We found that the source of the problem comes from these lines in `checker.go`, presumably occuring during mtail  parsing:

`20: const kMaxRegexpLen = 1024`
`51:	if c.depth > 100 {`

Bumping these values up (to 3072 and 200 respectively) and rebuilding mtail locally seems to solve these problems.

Its my understanding these were added for performance reasons. However, in our specific situation, performance is not a concern for us as we use mtail for testing purposes. 

We're left with a dilemma here since, as far as I can tell, there is no way these values are configurable given they are hardcoded values. 

The only solution we can think of that both addresses the performance concerns and our requirements, without forking the repository for our own purposes, is to make these values configurable. I wanted to run it by you guys and understand if there is appetite for making these values configurable via the command line interface. Or configurable some other method would be acceptable too (environment variables?). If so, I'll be willing to submit a PR to implement that functionality (assuming I can figure out how to do it 👯 ).

Long story short, I'm asking, if I make these values configurable, will you be willing to merge my changes? 

Thanks,
Danielle

**Comment by jaqx0r** [comment]
Gday. The limit is in place because the fuzzer found it could make the
compiler run forever on long input.   It needs to be able to terminate and
I thought 4k of text ought to be enough for anybody. :)

If you'd like to submit a PR, I'll accept it.
